zetaclient/zetabridge/zetacore_bridge.go

```diff
@@ -218,7 +218,7 @@ func (b *ZetaCoreBridge) UpdateZetaCoreContext(coreContext *corecontext.ZetaCore
 
 	// check and update chain params for each chain
 	for _, chainParam := range chainParams {
-		err := config.ValidateChainParams(chainParam)
+		err := observertypes.ValidateChainParams(chainParam)
```

Comment:
I think it would make sense to move the validation and the chain params structure to the common package. It would be better to keep zetaclient and zetacore binaries as independent of each other as possible

Reply:
I don't so for this one. I feel like we're putting too many things in the `common` package while it should be more used for general, generic data types.
Here `ValidateChainParams` is clearly part of the `observer` namespace since the type is defined and stored in this module. Putting the validation the validation logic outside is contrary to the idea of modularity of the SDK.
>It would be better to keep zetaclient and zetacore binaries as independent of each other as possible
ZetaClient must import `observertypes` in any case to interact with the client.

Another example is the header and proof objects that we should move from common to lightclient once the module is created.

Reply:
I feel like all types and related functions which are used by both the client and the core should live in the common package , if possible .

Just so that we have a clear separation between them.

Reply:
Yes, it theory it shouldn't be needed on ZetaClient as invalid chain params would be a broken invariant on the blockchain. But I think we can keep it as an extra measure.

>I feel like all types and related functions which are used by both the client and the core should live in the common package , if possible .

The ZetaClient depends on ZetaCore in any case since it must communicate through its API, I don't think there is an issue on this side. It's having ZetaCore having some dependencies on ZetaClient that would be an anti-pattern

Reply:
The validation is good to have on both. I am suggesting having the common as something that is imported by both the zetacore and zetaclient so that they can remain independent of each other. Which I believe suits the naming `common`. 

However, if that is not needed or desirable, we can import zetacore into the client .

Reply:
That seems fine to me; we can merge this for now and discuss later . 
Zetacore does import common packages already in some instances. For example, Chain is defined in common.  I cant think of any reason we would not be able to move other types to common as well